tests/PasswordHashingProvider.cpp

```diff
@@ -0,0 +1,185 @@
+#include <gtest/gtest.h>
+#include <string>
+#include "../src/PasswordHashingProvider.hpp"
+
+class PasswordHashingProviderTests : public ::testing::Test {
+public:
+    bool checkIfStringContainsSubstring(const std::string& parentString, const std::string& subString) {
+        return std::string::npos != parentString.find(subString);
+    }
+
+    const unsigned defaultSaltLength = 16;
+    const unsigned defaultHashLength = 16;
+    const std::string defaultSalt{"1234567890123456"};
+    const std::string defaultPassword{"trustnoone"};  
+};
+
+TEST_F(PasswordHashingProviderTests, WhenUsingDefaultConstructorSaltLengthIsEqualTo16)
+{
+    const unsigned expectedSaltLength = 16;
+    PasswordHashingProvider h1 = PasswordHashingProvider();
+        
+    std::string salt1 = h1.generateSalt();
+    
+    ASSERT_EQ(salt1.length(), expectedSaltLength);
+}
+
+TEST_F(PasswordHashingProviderTests, CanGenerateSaltWithGivenLength)
+{
+    const unsigned SaltLength = 24;
+    PasswordHashingProvider h1(SaltLength, defaultHashLength);
+    
+    std::string salt1 = h1.generateSalt();
+    
+    ASSERT_EQ(salt1.length(), SaltLength);
+}
+
+TEST_F(PasswordHashingProviderTests, CannotGenerateSaltShorterThan8)
+{
+    const unsigned givenSaltLength = 4;
+    const unsigned expectedSaltLength = 8;
+    PasswordHashingProvider h1(givenSaltLength, defaultHashLength);
+    
+    std::string salt1 = h1.generateSalt();
+    
+    ASSERT_EQ(salt1.length(), expectedSaltLength);
+}
+
+TEST_F(PasswordHashingProviderTests, CannotGenerateSaltLongerThan32)
+{
+    const unsigned givenSaltLength = 33;
+    const unsigned expectedSaltLength = 32;
+    PasswordHashingProvider h1(givenSaltLength, defaultSaltLength);
+    
+    std::string salt1 = h1.generateSalt();
+    
+    ASSERT_EQ(salt1.length(), expectedSaltLength);
+}
+
+TEST_F(PasswordHashingProviderTests, CanGenerateDifferentSaltsDuringFirstUseFromTwoInstances)
+{
+    PasswordHashingProvider h1(defaultSaltLength, defaultHashLength);
+    PasswordHashingProvider h2(defaultSaltLength, defaultHashLength);
+    
+    std::string salt1 = h1.generateSalt();
+    std::string salt2 = h2.generateSalt();
+
+    ASSERT_NE(salt1, salt2);
+}
+
+TEST_F(PasswordHashingProviderTests, CanGenerateDifferentSaltsFromOneInstance)
+{
+    PasswordHashingProvider h1(defaultSaltLength, defaultHashLength);
+        
+    std::string salt1 = h1.generateSalt();
+    std::string salt2 = h1.generateSalt();
+
+    ASSERT_NE(salt1, salt2);
+}
+
+TEST_F(PasswordHashingProviderTests, WhenUsingDefaultConstructorHashLengthIsEqualTo16)
+{
+    const unsigned givenHashLength = 16;
+    const unsigned expectedHashLength = 2*givenHashLength;
```

Comment:
givenHashLength is used only here. Is it needed? shouldn't it be defaultHashLength?

Reply:
Copy-paste driven test writing :)